src/person.controller.ts

```diff
@@ -21,14 +21,30 @@ export class PersonController {
     return this.personModel.findById(params.id).exec();
   }
 
+  @Delete(":id")
+  @HttpCode(200)
+  DeleteById(@Param() params: {id: string}): Promise<PersonModel>  {
+    return this.personModel.findByIdAndRemove(params.id).exec();
+  }
+
+  @Post(":id")
+  @HttpCode(200)
+  UpdateById(@Param() params: {id: string}): Promise<PersonModel> {
+    return this.personModel.findById(params.id).update(
```

Comment:
Co tu robisz... tłumaczę. Najpierw pobierasz dokument z bazy danych po jego ID, a następnie na tym dokumencie starasz się wykonać operację update. Nie sądzę, że to tak zadziała.

Idąc za dokumentacją https://mongoosejs.com/docs/api.html#model_Model-update
`update(query, updateDoc)` to jest zapis, który nas interesuje. Nie musisz wybierać dokumentu z bazy danych, bo:
- w pole query wstawiamy obiekt, który nam powie jakie dokumenty mają zostać zaktualizowane w bazie danych, w naszym przypadku chcemy zaktualizować wszystkie dokumenty, które mają zadane ID (czyli na dobrą sprawę dokładnie jeden),
- w pole updateDoc wstawiamy obiekt samego update - pomijając całokształt to jest dobrze zrobione.

Jak się z tym uporasz, to następnym krokiem jest, żeby body do update przekazywać tak samo jak do "create", czyli nie na sztywno jak teraz jest to zrobione.

Reply:
Aha, jeszcze warto dodać, że `update()` aktualizuje wyłącznie jeden dokument, więc jak mu podasz warunek, który znajduje więcej niż jeden, to zostanie zaktualizowany pierwszy jaki znajdzie. Do aktualizacji wielu na raz służy bardzo podobne `updateMany()`.